### PR TITLE
Upgrade to OmniAuth 2

### PR DESCRIPTION
**Pull request by 5andi5**

### CODE REVIEW

**Comment by 5andi5** [comment]
@ebeigarts Could You review, please?
I'm planning to update related strategy gems similary and probably to create new ones as well.

File: lib/omniauth/strategies/seb.rb [hunk 1]
```diff
@@ -62,24 +70,51 @@ def callback_phase
...
-          form.to_html.gsub('</form>', '</form><script type="text/javascript">document.forms[0].submit();</script>'))
+          form.to_html.gsub('</form>', "</form><script type=\"text/javascript\"#{nonce_attribute}>document.forms[0].submit();</script>"))
         form.to_response
       end
+
+      private
+
+      def set_locale_from_query_param
+        locale = request.params['locale']
+        if (locale != nil && locale.strip != '' && I18n.locale_available?(locale))
```
Comment (liepauls) [inline lib/omniauth/strategies/seb.rb]: Hmm, could probably just use `I18n.locale_available?`, no?

```ruby
I18n.locale_available? nil
=> false
I18n.locale_available? ' '
=> false
I18n.locale_available? 'asdf'
=> false
```

Tested with ruby 2.7, rails 6

File: lib/omniauth/strategies/seb/response.rb [hunk 2]
```diff
@@ -22,8 +22,12 @@ def validate!(pub_key)
           raw_str = SIGNED_KEYS.map{|k| prepend_length(@hash[k])}.join
           received_sig_str = Base64.decode64(@hash['IB_CRC'])
 
-          if !pub_key.verify(OpenSSL::Digest::SHA1.new, received_sig_str, raw_str)
-            raise ValidationError, 'Invalid signature'
+          if !pub_key.verify(OpenSSL::Digest::SHA512.new, received_sig_str, raw_str)
```
Comment (liepauls) [inline lib/omniauth/strategies/seb/response.rb:25]: ❤️

**Review by liepauls — APPROVED** [review]
Great! 👍

File: lib/omniauth/strategies/seb.rb (hunk 1 above)
Comment (5andi5) [reply lib/omniauth/strategies/seb.rb]: @liepauls Thanks - improved

**Review by ebeigarts — APPROVED** [review]

**Comment by ebeigarts** [comment]
@5andi5 thanks!

**Pull request merged**